integration-tests/tests/src/tests/results.ts

```diff
@@ -0,0 +1,1060 @@
+
+      this.realm.write(() => {
+        const objects = createTestObjects(10);
+        const snapshot = objects.snapshot();
+
+        this.realm.deleteAll();
+        expect(objects.length).equals(0);
+        expect(snapshot.length).equals(10);
+        expect(snapshot[0]).equals(null);
+      });
+
+      this.realm.write(() => {
+        const objects = createTestObjects(10);
+        this.realm.deleteAll();
+
+        const snapshot = objects.snapshot();
+        expect(objects.length).equals(0);
+        expect(snapshot.length).equals(0);
+      });
+
+      this.realm.write(() => {
+        const objects = createTestObjects(10);
+        const snapshot = objects.snapshot();
+
+        this.realm.delete(snapshot);
+        expect(objects.length).equals(0);
+        expect(snapshot.length).equals(10);
+        expect(snapshot[0]).equals(null);
+      });
+
+      this.realm.write(() => {
+        const objects = createTestObjects(10);
+        this.realm.delete(objects);
+
+        const snapshot = objects.snapshot();
+        expect(objects.length).equals(0);
+        expect(snapshot.length).equals(0);
+      });
+    });
+
+    it("should iterate", function (this: RealmContext) {
+      this.realm.write(() => {
+        this.realm.create("TestObject", { doubleCol: 2 });
+        this.realm.create("TestObject", { doubleCol: 3 });
+      });
+
+      const results = this.realm.objects<TestObject>("TestObject").filtered("doubleCol >= 2");
+      expect(results.length).equals(2);
+      let calls = 0;
+      for (const obj of results) {
+        this.realm.write(() => {
+          obj.doubleCol = 1;
+        });
+        calls++;
+      }
+      expect(results.length).equals(0);
+      expect(calls).equals(2);
+    });
+
+    it("should iterate correctly when deleting objects", function (this: RealmContext) {
+      this.realm.write(() => {
+        this.realm.create("TestObject", { doubleCol: 2 });
+        this.realm.create("TestObject", { doubleCol: 3 });
+      });
+
+      const results = this.realm.objects("TestObject");
+      expect(results.length).equals(2);
+      let calls = 0;
+      for (const obj of results) {
+        this.realm.write(() => {
+          this.realm.delete(obj);
+          calls++;
+        });
+      }
+      expect(calls).equals(2);
+      expect(this.realm.objects("TestObject").length).equals(0);
+    });
+  });
+
+  describe("Filtering and sorting", () => {
+    it("implements filtered", () => {
+      const realm = new Realm({ schema: [PersonObjectSchema, DefaultValuesSchema, TestObjectSchema] });
+
+      realm.write(function () {
+        realm.create("PersonObject", { name: "Ari", age: 10 });
+        realm.create("PersonObject", { name: "Tim", age: 11 });
+        realm.create("PersonObject", { name: "Bjarne", age: 12 });
+        realm.create("PersonObject", { name: "Alex", age: 12, married: true });
+      });
+
+      expect(realm.objects("PersonObject").filtered("truepredicate").length).equals(4);
```

Comment:
This is probably also covered in the query tests.  Might be able to refactor this out.